### PR TITLE
new port pffft

### PR DESCRIPTION
**Pull request by luadebug**
[ x ] Im confused how to deal with option here... 
[ ] License itself does not seem into repo. (Like License file is seems to be used to store into conan, and being stored into vcpkg repo itself)...
[ x ] Maybe there is better url to use.
[ ] De-bundle fftpack & if there is need & possible (most likely this is abandoned & modified bundled dependency)? Quite funny to see it still stored there (https://chromium.googlesource.com/chromium/src/+/c1086af075699b05612673ef0005b0d3e0817a04/third_party/pffft/src & here https://healpix.jpl.nasa.gov/html/libfftpack/dirs.html) (Most likely file header stores LICENSE itself.)
If everything is ok feel free to merge, if there is need edit feel free to edit.

### CODE REVIEW

**Comment by luadebug** [comment]
Is it ```set_license("BSD-3-Clause")```?

File: packages/p/pffft/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package("pffft")
+
+    set_homepage("https://bitbucket.org/jpommier/pffft/")
+    set_description("PFFFT, a pretty fast Fourier Transform.")
+    set_license("BSD-like (FFTPACK license)")
```
Comment (waruqi) [inline packages/p/pffft/xmake.lua:5]: add new line

File: packages/p/pffft/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    add_urls("https://bitbucket.org/jpommier/pffft.git")
```
Comment (waruqi) [inline packages/p/pffft/xmake.lua:6]: newline

File: packages/p/pffft/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    add_configs("nosimd", {description = "Build without SIMD support.", default = false, type = "boolean"})
```
Comment (waruqi) [inline packages/p/pffft/xmake.lua:7]: remove `no` prefix, you can use `simd` and true as default value.

File: packages/p/pffft/port/xmake.lua [hunk 4]
```diff
@@ -0,0 +1,20 @@
+add_rules("mode.debug", "mode.release")
+
+option("simd")
+    set_default(false)
```
Comment (waruqi) [inline packages/p/pffft/port/xmake.lua:4]: true

File: packages/p/pffft/port/xmake.lua [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    set_description("Build without SIMD support.")
```
Comment (waruqi) [inline packages/p/pffft/port/xmake.lua:5]: with

File: packages/p/pffft/port/xmake.lua [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    add_defines("PFFFT_SIMD_DISABLE")
```
Comment (waruqi) [inline packages/p/pffft/port/xmake.lua:6]: remove it. then

```lua
target("...")
    if not has_config("simd") then
        add_defines("PFFFT_SIMD_DISABLE")
    end

File: packages/p/pffft/port/xmake.lua [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+target("pffft")
+    set_kind("$(kind)")
+    add_options("nosimd")
+    if is_kind("shared") then
```
Comment (waruqi) [inline packages/p/pffft/port/xmake.lua:11]: and `is_plat("windows")`

File: packages/p/pffft/port/xmake.lua [hunk 8]
```diff
@@ -0,0 +1,17 @@
+add_rules("mode.debug", "mode.release")
+
+target("pffft")
+    set_kind("$(kind)")
+    if not has_config("simd") then
```
Comment (waruqi) [inline packages/p/pffft/port/xmake.lua:5]: you need define option, do not remove it.

File: packages/p/pffft/xmake.lua [hunk 9]
```diff
@@ -0,0 +1,24 @@
+package("pffft")
+
+    set_homepage("https://bitbucket.org/jpommier/pffft/")
+    set_description("PFFFT, a pretty fast Fourier Transform.")
+    set_license("BSD-like (FFTPACK license)")
+
+    add_urls("https://bitbucket.org/jpommier/pffft.git")
```
[the remark this answers is not in the document]
Comment (luadebug) [reply packages/p/pffft/xmake.lua:7]: Indeed, I do not find anything else than ```https://bitbucket.org/jpommier/pffft/get/02fe7715a5bf.zip```

File: packages/p/pffft/xmake.lua [hunk 10]
```diff
@@ -0,0 +1,25 @@
+package("pffft")
+
+    set_homepage("https://bitbucket.org/jpommier/pffft/")
+    set_description("PFFFT, a pretty fast Fourier Transform.")
+    set_license("BSD-like (FFTPACK license)")
+
+    add_urls("https://bitbucket.org/jpommier/pffft/$(version).tar.gz",
+             "https://bitbucket.org/jpommier/pffft.git")
+    add_versions("02fe7715a5bf8bfd914681c53429600f94e0f536", "9adeb18ac7bb52e9fb921c31c0c6a4e9ae150cc6fcb20a899d4b3a2275176ded")
```
Comment (waruqi) [inline packages/p/pffft/xmake.lua:9]: only support semver, you need map it to commit in version function. see other packages. e.g. sqlite3

and add a new line

Comment (luadebug) [reply packages/p/pffft/xmake.lua:9]: https://github.com/microsoft/vcpkg/blob/80d54ff62d528339c626a6fbc3489a7f25956ade/ports/pffft/portfile.cmake#L6-L8
https://github.com/conan-io/conan-center-index/blob/4622ac85d1cec8cb7a2fcc8a1796d4b73bff285e/recipes/pffft/all/conandata.yml#L1-L7
We could go back into 
```lua
add_urls("https://bitbucket.org/jpommier/pffft/get/$(version).zip")
add_versions("02fe7715a5bf", "81e1b91bad77092681e5ed6c2eef1a3bae7eb2154d3358a10c663867cd947396")
```

**Pull request merged**